processes/metaphlan.nf

```diff
@@ -4,31 +4,40 @@ process metaphlan {
 
     input:
     tuple val(sample), path(kneads)
-    path unmatched
     path metaphlan_db
 
     output:
     val  sample                  , emit: sample
     path "${sample}_profile.tsv" , emit: profile
-    path "${sample}_grouped.fastq.gz"
+    // path "${sample}_grouped.fastq.gz"
     path "${sample}_bowtie2.tsv"
     path "${sample}.sam"
 
     script:
-    def forward = kneads[0]
-    def reverse = kneads[1]
-    def unf = unmatched[0]
-    def unr = unmatched[1]
+    // Determine if there is one or four files based on the input file names
+    def kneads_files = kneads.size() == 1 ? "single" : "paired"
 
     """
-    cat $forward $reverse $unf $unr > ${sample}_grouped.fastq.gz
-    
-    metaphlan ${sample}_grouped.fastq.gz ${sample}_profile.tsv \
-        --bowtie2out ${sample}_bowtie2.tsv \
-        --samout ${sample}.sam \
-        --input_type fastq \
-        --nproc ${task.cpus} \
-        --bowtie2db $metaphlan_db
+    // For paired-end, concatenate files before running metaphlan
+    if [ "$kneads_files" == "paired" ]; then
+        cat ${kneads[0]} ${kneads[1]} ${kneads[2]} ${kneads[3]} > ${sample}_grouped.fastq.gz
+
+        metaphlan ${sample}_grouped.fastq.gz ${sample}_profile.tsv \
+            --bowtie2out ${sample}_bowtie2.tsv \
+            --samout ${sample}.sam \
+            --input_type fastq \
+            --nproc ${task.cpus} \
+            --bowtie2db $metaphlan_db
+
+    else
+
+        metaphlan ${kneads[0]}_grouped.fastq.gz ${sample}_profile.tsv \
+            --bowtie2out ${sample}_bowtie2.tsv \
+            --samout ${sample}.sam \
+            --input_type fastq \
+            --nproc ${task.cpus} \
+            --bowtie2db $metaphlan_db
+    fi
```

Comment:
We should not use shell logic for this, we should use groovy. And in any case, we don't want to duplicate the majority of the command. Could maybe do something like

```
def shouldcat = kneads.size() > 1 ? "cat $kneads > ${sample}_grouped.fastq.gz : "echo 'skipping cat'"
def inputline = kneads.size() > 1 ? "metaphlan ${sample}_grouped.fastq.gz : "metaphlan $kneads

"""
$shouldcat
$inputline {sample}_profile.tsv \
...
```